.github/workflows/deploy_doc.yml

```diff
@@ -4,11 +4,45 @@ name: Deploy docs
 on: 
   workflow_dispatch:
   push:
-      branches: [master]
+      branches: [master,versions] # TODO: remove branch versions 
+      tags:
   
 jobs:
+  evaluate-label:
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.label_step.outputs.version}}
+    steps:
+      - run: echo ${{ github.event_name }} #TODO: remove this debugging line 
+      - run: echo ${{ github.ref }} # TODO: remove this debugging line 
+      - name: checks for the label 
+        id: label_step 
+        run: |
+          if [[ "${{ github.event_name }}" == "push" ]] && [[ "${{ github.ref  }}" == "refs/heads/versions" ]]; then
+                echo "version=latest" >> $GITHUB_OUTPUT
+          fi
+          if [[ "${{ github.event_name }}" == "push" ]] && [[ "${{ github.ref  }}" != "refs/heads/versions" ]]; then
+                echo "version=Error" >> $GITHUB_OUTPUT
+          fi
+          if [[ "${{ github.event_name }}" == "tag" ]]; then
+                echo "version=stable" >> $GITHUB_OUTPUT
+          fi 
+          
+  # TODO: this is a debugging job, delete it 
+  generate-docs:
+    needs: [evaluate-label]
+    if: needs.evaluate-label.outputs.label != 'Error'
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo ${{needs.evaluate-label.outputs.label}} 
 
   deploy-docs:
-    uses: qibogang/workflows/.github/workflows/deploy-sphinx-docs-to-ghpages.yml@cache_pip
+    needs: [evaluate-label]
+    if: needs.evaluate-label.outputs.label != 'Error'
+    uses: qibogang/workflows/.github/workflows/deploy-sphinx-docs-to-ghpages.yml@versions
```

Comment:
You can't use this action here, unfortunately for the time being you have to copy-paste the content above.